sklearn/datasets/_base.py

```diff
@@ -99,6 +100,19 @@ def _convert_data_dataframe(
     return combined_df, X, y
 
 
+@validate_params(
+    {
+        "container_path": [str, Path],
```

Comment:
```suggestion
        "container_path": [str, os.PathLike],
```

Please also fix the parameter description in the docstring to ``str or path-like``